iree/compiler/Dialect/HAL/Target/LLVM/internal/UnixLinkerTool.cpp

```diff
@@ -81,28 +81,37 @@ class UnixLinkerTool : public LinkerTool {
 
     SmallVector<std::string, 8> flags = {
         getToolPath(),
-
-        // Avoids including any libc/startup files that initialize the CRT as
-        // we don't use any of that. Our shared libraries must be freestanding.
-        "-nostdlib",  // -nodefaultlibs + -nostartfiles
-
-        // Statically link all dependencies so we don't have any runtime deps.
-        // We cannot have any imports in the module we produce.
-        // "-static",
-
-        // HACK: we insert mallocs and libm calls. This is *not good*.
-        // We need hermetic binaries that pull in no imports; the MLIR LLVM
-        // lowering paths introduce a bunch, though, so this is what we are
-        // stuck with.
-        "-shared",
-        "-undefined suppress",
-
         "-o " + artifacts.libraryFile.path,
     };
 
     if (targetTriple.isOSDarwin() || targetTriple.isiOS()) {
+      // Statically link all dependencies so we don't have any runtime deps.
```

Comment:
these seem unrelated? possibly due to merge issues?

Reply:
Yes sorry, I thought a PR code review would only take a day or two. Scott wanted me to merge in his changes so that I could successfully build, and this ended up piggy backing. I wasn't sure how to isolate the first set of changes after I had already submitted the PR. I will be aware of this possibility in the future.

Reply:
A few options:

A) Merge `google:main` into your PR branch (which ideally wouldn't also be `main`, but too late to change that now :P). This will preserve all history.
B) Rebase your PR branch onto `google:main`, editing history (e.g. dropping `astrotuna proposed MacOS linker problem fix` and `update linker settings for macOS`, or keeping those commits and resolving the merge conflicts with the similar changes made to `google:main`)

Ultimately, you should either fixup the history on your PR _or_ merge with "squash and merge". Both are fine.

If there are actually more changes you need made to `UnixLinkerTool.cpp` for Apple platforms, those should be made on a separate PR.